generator/internal/genclient/translator/protobuf/protobuf_test.go

```diff
@@ -0,0 +1,255 @@
+			Name:          "Fake",
+			Documentation: "A test message.",
+			Fields: []*genclient.Field{
+				{Repeated: true, Documentation: "A repeated field tag = 1", Name: "f_double", JSONName: "fDouble", ID: ".test.Fake.f_double", Typez: genclient.DOUBLE_TYPE},
+				{Repeated: true, Documentation: "A repeated field tag = 3", Name: "f_int64", JSONName: "fInt64", ID: ".test.Fake.f_int64", Typez: genclient.INT64_TYPE},
+				{Repeated: true, Documentation: "A repeated field tag = 9", Name: "f_string", JSONName: "fString", ID: ".test.Fake.f_string", Typez: genclient.STRING_TYPE},
+				{Repeated: true, Documentation: "A repeated field tag = 12", Name: "f_bytes", JSONName: "fBytes", ID: ".test.Fake.f_bytes", Typez: genclient.BYTES_TYPE},
+			},
+		})
+	} else {
+		t.Errorf("Cannot find message %s in API State", "Fake")
+	}
+}
+
+func TestScalarOptional(t *testing.T) {
+	contents := `
+syntax = "proto3";
+package test;
+
+// A test message.
+message Fake {
+    // An optional field tag = 1
+    optional double   f_double     = 1;
+		
+	// An optional field tag = 3
+    optional int64    f_int64      = 3;
+	
+	// An optional field tag = 9
+    optional string   f_string     = 9;
+	
+	// An optional field tag = 12
+    optional bytes    f_bytes      = 12;
+}
+`
+	translator := NewTranslator(&Options{
+		Request:     newCodeGeneratorRequest("resources.proto", contents, t),
+		Language:    "not used",
+		OutDir:      "not used",
+		TemplateDir: "not used",
+	})
+
+	api := translator.makeAPI()
+
+	if message := api.State.MessageByID[".test.Fake"]; message != nil {
+		checkMessage(t, *message, genclient.Message{
+			Name:          "Fake",
+			Documentation: "A test message.",
+			Fields: []*genclient.Field{
+				{Optional: true, Documentation: "An optional field tag = 1", Name: "f_double", JSONName: "fDouble", ID: ".test.Fake.f_double", Typez: genclient.DOUBLE_TYPE},
+				{Optional: true, Documentation: "An optional field tag = 3", Name: "f_int64", JSONName: "fInt64", ID: ".test.Fake.f_int64", Typez: genclient.INT64_TYPE},
+				{Optional: true, Documentation: "An optional field tag = 9", Name: "f_string", JSONName: "fString", ID: ".test.Fake.f_string", Typez: genclient.STRING_TYPE},
+				{Optional: true, Documentation: "An optional field tag = 12", Name: "f_bytes", JSONName: "fBytes", ID: ".test.Fake.f_bytes", Typez: genclient.BYTES_TYPE},
+			},
+		})
+	} else {
+		t.Errorf("Cannot find message %s in API State", "Fake")
+	}
+}
+
+func newCodeGeneratorRequest(name, contents string, t *testing.T) *pluginpb.CodeGeneratorRequest {
+	accessor := protocompile.SourceAccessorFromMap(map[string]string{
+		name: contents,
+	})
+	compiler := protocompile.Compiler{
+		Resolver:       &protocompile.SourceResolver{Accessor: accessor},
+		MaxParallelism: 1,
+		SourceInfoMode: protocompile.SourceInfoStandard,
+	}
+	files, err := compiler.Compile(context.Background(), name)
+	if err != nil {
+		t.Errorf("Error compiling proto %q", err)
```

Comment:
`t.Fatalf` since I believe you want the test to stop executing at this point

also nit: error messages are typically lowercase in Go

Reply:
Done.